From a0d57bebb3a5d7259a0508a07a898cfe044155a1 Mon Sep 17 00:00:00 2001
From: Lawrence D'Oliveiro <ldo@geek-central.gen.nz>
Date: Tue, 4 Jul 2017 06:56:00 +0000
Subject: [PATCH 07/12] No longer automatically build with ImageMagick or
 GraphicsMagick if present; now need to explicitly specify --with-imagemagick
 or --with-graphicsmagick to configure

---
 ChangeLog    |  5 +++++
 configure.ac | 30 ++++++++++++++++++------------
 2 files changed, 23 insertions(+), 12 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 596e22d..1cf42b1 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+0.7.2+: 2017 July 4
+    No longer automatically build with ImageMagick or GraphicsMagick if
+	    present; now need to explicitly specify --with-imagemagick or
+		--with-graphicsmagick to configure
+
 0.7.2: 2016 December 31
 	Various code-quality and build improvements
 	Support “jump pgc n” and other interaction code-generation improvements
diff --git a/configure.ac b/configure.ac
index d65cc32..1b7ec6f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -45,21 +45,27 @@ if test "$enable_shared" = 'no'; then
     config_static='--static'
 fi
 
-usemagick=0
+use_imagemagick=0
+use_graphicsmagick=0
+AC_ARG_WITH([imagemagick], AS_HELP_STRING([--with-imagemagick], [Use ImageMagick to augment image import formats]), [use_imagemagick=1])
+AC_ARG_WITH([graphicsmagick], AS_HELP_STRING([--with-graphicsmagick], [Use GraphicsMagick to augment image import formats]), [use_graphicsmagick=1])
+if test "$use_imagemagick" = 1 && test "$use_graphicsmagick" = 1; then
+    AC_MSG_ERROR([cannot specify both --with-imagemagick and --with-graphicsmagick], 1)
+fi
 
-PKG_CHECK_MODULES([IMAGEMAGICK], [ImageMagick >= 5.5.7], usemagick=1; AC_DEFINE(HAVE_MAGICK, 1, [Whether the ImageMagick libraries are available]), [:])
-if test "$usemagick" = 1; then
+if test "$use_imagemagick" = 1; then
+    PKG_CHECK_MODULES([IMAGEMAGICK], [ImageMagick >= 5.5.7], [AC_DEFINE(HAVE_MAGICK, 1, [Whether the ImageMagick libraries are available])], [AC_MSG_ERROR([ImageMagick not available])])
     MAGICK_CFLAGS="$IMAGEMAGICK_CFLAGS"
     MAGICK_LIBS="$IMAGEMAGICK_LIBS"
-else
-    PKG_CHECK_MODULES([GRAPHICSMAGICK], [GraphicsMagick], usemagick=1; [AC_DEFINE(HAVE_GMAGICK, 1, [whether the GraphicsMagick libraries are available])], [:])
-    if test "$usemagick" = 1; then
-        MAGICK_CFLAGS="$GRAPHICSMAGICK_CFLAGS"
-        MAGICK_LIBS="$GRAPHICSMAGICK_LIBS"
-    else
-        MAGICK_CFLAGS="$LIBPNG_CFLAGS"
-        MAGICK_LIBS="$LIBPNG_LIBS"
-    fi
+fi
+if test "$use_graphicsmagick" = 1; then
+    PKG_CHECK_MODULES([GRAPHICSMAGICK], [GraphicsMagick], [AC_DEFINE(HAVE_GMAGICK, 1, [whether the GraphicsMagick libraries are available])], [AC_MSG_ERROR([GraphicsMagick not available])])
+    MAGICK_CFLAGS="$GRAPHICSMAGICK_CFLAGS"
+    MAGICK_LIBS="$GRAPHICSMAGICK_LIBS"
+fi
+if test "$use_imagemagick" != 1 && test "$use_graphicsmagick" != 1; then
+    MAGICK_CFLAGS="$LIBPNG_CFLAGS"
+    MAGICK_LIBS="$LIBPNG_LIBS"
 fi
 
 AC_SUBST(MAGICK_CFLAGS)
-- 
2.51.0

